sqlglot/parser.py

```diff
@@ -2452,6 +2455,25 @@ def _parse_subquery(
             alias=self._parse_table_alias() if parse_alias else None,
         )
 
+    def _implicit_unnests_to_explicit(self, this: E) -> E:
+        refs = {this.args["from"].this.alias_or_name.lower()}
+        for i, join in enumerate(this.args.get("joins") or []):
+            table = join.this
+            if isinstance(table, exp.Table) and not join.args.get("on"):
+                if table.parts[0].name.lower() in refs:
```

Comment:
why are you using name.lower, shouldn't we normalize the names with the dialect?

Reply:
we probably should, though note that this is how it was previously implemented in redshift

Reply:
yea but redshift, everything is lower case so that makes sense :)

Reply:
ah lol, 2 steps ahead

Reply:
makes sense - will refactor thanks for catching

Reply:
There's an annoying edge case in BQ if we normalize the identifiers.. we currently use a [heuristic](https://github.com/tobymao/sqlglot/blob/984049648991b8b1f89827ec35b29da603e6fb64/sqlglot/dialects/bigquery.py#L272) to check whether a table ref. should be normalized which is to check if it's qualified. The reason is that CTEs are case-insensitive while tables usually aren't. This breaks the following test:

```sql
SELECT results FROM Coordinates, Coordinates.position AS results;
SELECT results FROM c.db.Coordinates AS Coordinates, UNNEST(Coordinates.position) AS results;
```

because the first `Coordinates` ref. gets normalized, since it doesn't have a `db` set, but the second `Coordinates` in `Coordinates.position` _doesn't_, because it _does_ have a `db` set. So the two don't match when we check the `refs` set and it's not converted into a column.